### PR TITLE
Fix:service detail res

### PR DESCRIPTION
**Pull request by ikaratass**
fix [AB#2821](https://dev.azure.com/switch-ev/09ab882d-ca4b-4b5b-b5b8-80603c12e5b2/_workitems/edit/2821) 

checks if the requested services ids in ServiceDetailReq are ids that were offered by the SECC in ServiceDiscoveryRes. if not, FAILED_SERVICE_ID_INVALID is returned.

### CODE REVIEW

File: iso15118/secc/states/iso15118_2_states.py [hunk 1]
```diff
@@ -412,6 +412,24 @@ async def process_message(
...
+            # [V2G2-425] The SECC shall respond with a negative ResponseCode
+            # 'FAILED_ServiceIDInvalid'
+            # if the EVCC provided a not previously retrieved ServiceID in the
+            # ServiceDetailsReq.
+            self.stop_state_machine(
+                f"[V2G2-425] {service_detail_req.service_id} is not offered",
+                message,
+                ResponseCode.FAILED_SERVICE_ID_INVALID,
+            )
+            return
```
Comment (tropxy) [inline iso15118/secc/states/iso15118_2_states.py:431]: This is no reason to stop the communication. we just need to communicate that that service ID is invalid and continue waiting for the next ServiceDetailReq or the next state

**Review by tropxy — CHANGES_REQUESTED** [review]

File: iso15118/secc/states/iso15118_2_states.py (hunk 1 above)
Comment (ikaratass) [reply iso15118/secc/states/iso15118_2_states.py]: Done

File: iso15118/secc/states/iso15118_2_states.py [hunk 2]
```diff
@@ -408,6 +409,28 @@ async def process_message(
...
+            if service_detail_req.service_id == service_details.service_id:
+                is_found = True
+                break
+
+        if not is_found:
+            # [V2G2-425] The SECC shall respond with a negative ResponseCode
+            # 'FAILED_ServiceIDInvalid'
+            # if the EVCC provided a not previously retrieved ServiceID in the
+            # ServiceDetailsReq.
+            msg_type = get_msg_type(str(message))
```
Comment (shalinnijel2) [inline iso15118/secc/states/iso15118_2_states.py:423]: As discussed, this is perhaps simpler. Also returns the service id of the failed ServiceDetailReq

```
            error_service_detail_res = ServiceDetailRes(
                response_code=ResponseCode.FAILED_SERVICE_ID_INVALID,
                service_id=service_detail_req.service_id,
            )
            self.create_next_message(
                None,
                error_service_detail_res,
                Timeouts.V2G_SECC_SEQUENCE_TIMEOUT,
                Namespace.ISO_V2_MSG_DEF,
            )
            return
```

Comment (ikaratass) [reply iso15118/secc/states/iso15118_2_states.py]: Done

**Review by shalinnijel2 — APPROVED** [review]

**Review by tropxy — APPROVED** [review]

**Review by shalinnijel2 — APPROVED** [review]

**Pull request merged**